src/main/java/ch/astorm/jotlmsg/io/FlatEntryListStructure.java

```diff
@@ -0,0 +1,147 @@
+package ch.astorm.jotlmsg.io;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Represents an [MS-OXCDATA] 2.3.3 FlatEntryList Structure in Java.
+ * 
+ * @author Guido Stein
+ *
+ */
+public class FlatEntryListStructure<T extends FlatEntryStructure> implements Iterable<T> {
+    private long count; // Number of FlatEntryStructures.
+    private long size; // Number of bytes of all FlatEntryStructures.
+    private List<T> flatEntryStructures = new ArrayList<T>();
+    private FlatEntryStructureFactory<T> fesf = new FlatEntryStructureFactory<T>();
+
+    /**
+     * Returns the number of FlatEntryStructures.
+     * 
+     * @return Number of structures.
+     */
+    public long getCount() {
+        return count;
+    }
+
+    /**
+     * Returns the number of total bytes of all FlatEntryStructures.
+     * 
+     * @return Number of total bytes.
+     */
+    public long getSize() {
+        return size;
+    }
+
+    /**
+     * Returns a list of FlatEntryStructures or subclass structures.
+     * 
+     * @return List of structures.
+     */
+    public List<T> getFlatEntryStructures() {
+        return flatEntryStructures;
+    }
+
+    /**
+     * Sets a list of FlatEntryStructures.
+     * 
+     * @param flatEntries List of FlatEntryStructures or subclass structures to set.
+     */
+    public void setFlatEntryStructures(List<T> flatEntries) {
+        this.flatEntryStructures = flatEntries;
+    }
+
+    /**
+     * Adds a FlatEntryStructure.
+     * 
+     * @param flatEntry FlatEntryStructure or subclass to add.
+     */
+    public void addFlatEntryStructure(T flatEntry) {
+        flatEntryStructures.add(flatEntry);
+
+        long totalSize = 0;
+
+        for (FlatEntryStructure fes : flatEntryStructures) {
+            totalSize += fes.getSize();
+            totalSize += 4; // 4 bytes variable size
+
+            // Consider 4 byte alignment
+            totalSize = (totalSize + 4) & ~3;
+        }
+
+        size = totalSize;
+        count = (long) flatEntryStructures.size();
+    }
+
+    /**
+     * Constructor, creates an empty java list representation.
+     * 
+     */
+    public FlatEntryListStructure() {
+    };
```

Comment:
The whole constructor can be entirely removed.

Reply:
I hava tried to remove it but it seems that the generic stuff doesn't work without it.